arduino/libraries/librariesmanager/install.go

```diff
@@ -126,11 +130,29 @@ func (lm *LibrariesManager) InstallZipLib(ctx context.Context, archivePath strin
 		return fmt.Errorf("archive is not valid: multiple files found in zip file top level")
 	}
 
-	libraryName := paths[0].Base()
+	extractionPath := paths[0]
+	libraryName := extractionPath.Base()
 	installPath := libsDir.Join(libraryName)
 
-	// Deletes libraries folder if already installed
-	if _, ok := lm.Libraries[libraryName]; ok {
+	if err := installPath.Parent().MkdirAll(); err != nil {
+		return err
+	}
+	defer func() {
+		// Clean up install dir if installation failed
+		files, err := installPath.Parent().ReadDir()
+		if err != nil {
+			return
+		}
+		if len(files) == 0 {
+			installPath.Parent().RemoveAll()
+		}
+	}()
```

Comment:
IIUC:
```
libsDir              == "/home/user/sketchbook/libraries"
installPath          == "/home/user/sketchbook/libraries/LIBNAME"
installPath.Parent() == "/home/user/sketchbook/libraries"
```

so instead of `installPath.Parent()` you can use `libsDir` that makes the goal of this piece of code much more explicit: create the `libraries` directory if not exist and remove it when it's no more needed

Reply:
Looking at this again I think I made a mistake actually, I don't want to verify if `/home/user/sketchbook/libraries` is empty but `/home/user/sketchbook/libraries/LIBNAME`. So `installPath` is correct but I don't have to get its parent. :thinking: